openidconnect/lib/src/models/requests/revoke_token_request.dart

```diff
@@ -6,17 +6,25 @@ class RevokeTokenRequest {
   final OpenIdConfiguration configuration;
   final String token;
   final TokenType tokenType;
+  final String? clientId;
+  final String? clientSecret;
 
   const RevokeTokenRequest({
     required this.configuration,
     required this.token,
     required this.tokenType,
+    this.clientId,
+    this.clientSecret,
   });
 
-  Map<String, String> toMap() => {
-        "token": token,
-        "token_type_hint": tokenType == TokenType.accessToken
-            ? "access_token"
-            : "refresh_token",
-      };
+  Map<String, String> toMap() {
+    var map = {"token": token,
+      "token_type_hint": tokenType == TokenType.accessToken
+          ? "access_token"
+          : "refresh_token",
+    };
+    if (clientId != null) map = {"client_id": clientId!, ...map};
```

Comment:
why not just add these in the original map ?
```dart
 var map = {"token": token,
      "token_type_hint": tokenType == TokenType.accessToken
          ? "access_token"
          : "refresh_token",
       if (clientId != null) "client_id": clientId,
       if (clientSecret != null) "client_secret": clientSecret
    };
```

Reply:
@ahmednfwela thanks for the feedback, I just did it the way it was done everywhere else in this codebase. :)

Let me know if I need to change it.

Reply:
FYI, the reason I hid the original map is specifically to protect against inadvertent overwriting of the properties that have to be there.

Reply:
I did a major refactor for readability, and changed the entire package to use this pattern
```dart
var map = {
      "token": token,
      "token_type_hint": tokenType == TokenType.accessToken
          ? "access_token"
          : "refresh_token",
       if (clientId != null) "client_id": clientId,
       if (clientSecret != null) "client_secret": clientSecret
    };
```
so please change it in the code

Reply:
@ahmednfwela hello, sorry but it looks like this PR has been merged and closed.

Reply:
no problem 👍 will do the refactor myself